src/jvm/main/org/jetbrains/kotlinx/lincheck/Result.kt

```diff
@@ -24,36 +24,30 @@ import kotlin.coroutines.*
  * the execution thread was suspended without any chance to be resumed,
  * meaning that all other execution threads completed their execution or were suspended too.
  */
-sealed class Result {
-    abstract val wasSuspended: Boolean
-    protected val wasSuspendedPrefix: String get() = (if (wasSuspended) "SUSPENDED + " else "")
-}
+sealed interface Result
 
 /**
  * Type of result used if the actor invocation returns any value.
  */
-data class ValueResult @JvmOverloads constructor(val value: Any?, override val wasSuspended: Boolean = false) : Result() {
-    override fun toString() = wasSuspendedPrefix + "$value"
+data class ValueResult @JvmOverloads constructor(val value: Any?) : Result {
+    override fun toString() = "$value"
 }
 
 /**
  * Type of result used if the actor invocation does not return value.
  */
-object VoidResult : Result() {
-    override val wasSuspended get() = false
-    override fun toString() = wasSuspendedPrefix + VOID
+object VoidResult : Result {
+    override fun toString() = VOID
 }
 
-object SuspendedVoidResult : Result() {
-    override val wasSuspended get() = true
-    override fun toString() = wasSuspendedPrefix + VOID
+object SuspendedVoidResult : Result {
+    override fun toString() = VOID
 }
 
 private const val VOID = "void"
 
-object Cancelled : Result() {
-    override val wasSuspended get() = true
-    override fun toString() = wasSuspendedPrefix + "CANCELLED"
+object Cancelled : Result {
+    override fun toString() ="CANCELLED"
```

Comment:
Should it remain SUSPENDED + CANCELLED? What are the pros and cons?

Reply:
I don't see anything useful in printing "SUSPENDED +" at all, because this information can be extracted from the trace. In the results it looks redundant and may confuse a user.